shell/common/shell.cc

```diff
@@ -1025,13 +1024,22 @@ void Shell::OnAnimatorDraw(fml::RefPtr<Pipeline<flutter::LayerTree>> pipeline,
     }
   }
 
+  size_t max_bytes = resource_cache_max_bytes_update_;
+  resource_cache_max_bytes_update_ = 0;
+
   task_runners_.GetRasterTaskRunner()->PostTask(
       [&waiting_for_first_frame = waiting_for_first_frame_,
        &waiting_for_first_frame_condition = waiting_for_first_frame_condition_,
-       rasterizer = rasterizer_->GetWeakPtr(),
-       pipeline = std::move(pipeline)]() {
+       rasterizer = rasterizer_->GetWeakPtr(), max_bytes,
+       pipeline = std::move(pipeline), this]() {
```

Comment:
I think initializing the discard callback outside this lambda could make it easier to read and could avoid capturing `this` here.

```
Rasterizer::DiscardCallback discard_callback =
      [&resize_mutex = resize_mutex_,
       &expected_frame_size = expected_frame_size_](flutter::LayerTree& tree) {
        std::scoped_lock<std::mutex> lock(resize_mutex);
        return tree.frame_size() != expected_frame_size;
      };
```

And then passing `&discard_callback = discard_callback` rather than `this`.

Reply:
The capture should be then discard_callback = std::move(discard_callback). I'm not sure why capturing resize_mutex_ and expected_frame_size_ individually though instead of just capturing this in discard_callback.